code/ui/blocks/src/blocks/Canvas.tsx

```diff
@@ -36,31 +115,120 @@ const usePreviewProps = (
   const stories = useStories(storyIds, docsContext);
   const isLoading = stories.some((s) => !s);
   const sourceProps = useSourceProps(
-    mdxSource ? { code: decodeURI(mdxSource) } : { ids: storyIds },
+    mdxSource ? { code: decodeURI(mdxSource), of: props.of } : { ids: storyIds, of: props.of },
     docsContext,
     sourceContext
   );
   if (withSource === SourceState.NONE) {
     return { isLoading, previewProps: props };
   }
 
+  // if the user has specified a layout prop, use that...
+  let layout = layoutProp;
+  // ...otherwise, try to infer it from the children 'parameters' prop
+  Children.forEach(children, (child) => {
+    if (layout) {
+      return;
+    }
+    layout = (child as ReactElement)?.props?.parameters?.layout;
+  });
```

Comment:
```suggestion
  const layout ??= Children.map(children, (child) => (child as ReactElement)?.props?.parameters?.layout;
  ).find(Boolean);
```

Reply:
I'm sorry I have to disagree here.
cleverness++, readability--